aptible/resource_app.go

```diff
@@ -641,72 +641,91 @@ func scaleServices(d *schema.ResourceData, meta interface{}) error {
-
+		// https://stackoverflow.com/a/74383278
+		service := s
+		serviceInterface := service.(map[string]interface{})
+		log.Printf("serviceInterface: %+v", serviceInterface)
 		// Find the corresponding old service
 		var oldServiceData map[string]interface{}
 		for _, oldS := range oldService.(*schema.Set).List() {
 			oldService := oldS.(map[string]interface{})
-			if oldService["process_type"].(string) == processType {
+			if oldService["process_type"].(string) == serviceInterface["process_type"].(string) {
 				oldServiceData = oldService
 				break
 			}
 		}
-
-		// Check if there are changes to other keys, ignoring `force_zero_downtime` and `simple_health_check`
-		hasOtherChanges := false
-		for key, newValue := range serviceData {
+		shouldScale := false
+		for key, newValue := range serviceInterface {
 			if key == "force_zero_downtime" || key == "simple_health_check" || key == "autoscaling_policy" || key == "service_sizing_policy" {
 				continue // Skip checking these keys. Nothing to do with manual scaling
 			}
 
 			if oldServiceData == nil || oldServiceData[key] != newValue {
-				hasOtherChanges = true
+				shouldScale = true
 				break
 			}
 		}
-
-		if hasOtherChanges {
-			filteredServices = append(filteredServices, serviceData)
+		if !shouldScale {
+			return nil
 		}
-	}
 
-	for _, s := range filteredServices {
-		// https://stackoverflow.com/a/74383278
-		service := s
-		serviceInterface := service.(map[string]interface{})
 		g.Go(func() error {
-			memoryLimit := int64(serviceInterface["container_memory_limit"].(int))
-			containerProfile := serviceInterface["container_profile"].(string)
-			containerCount := int64(serviceInterface["container_count"].(int))
+			memoryLimit := serviceInterface["container_memory_limit"]
+			containerProfile := serviceInterface["container_profile"]
+			containerCount := serviceInterface["container_count"]
 			processType := serviceInterface["process_type"].(string)
 
 			log.Printf(
 				"Updating %s service to count: %d, limit: %d, and container profile: %s\n",
 				processType, containerCount, memoryLimit, containerProfile,
 			)
-			service, err := client.GetServiceForAppByName(appID, processType)
-			if err != nil {
-				log.Println("There was an error when finding the service \n[ERROR] -", err)
-				return generateErrorFromClientError(err)
+			service := findApiServiceByName(apiServices, processType)
+			if service == nil {
+				return fmt.Errorf("there was an error when finding the service: %s", processType)
+			}
+
+			payload := aptibleapi.NewCreateOperationRequest("scale")
+			if oldServiceData["container_count"] != containerCount && containerCount != -1 {
+				log.Println("Updating containers-count")
+				payload.SetContainerCount(int32(containerCount.(int)))
 			}
-			err = client.ScaleService(service.ID, containerCount, memoryLimit, containerProfile)
+			log.Printf("memory limit old: %s, new: %s", oldServiceData["container_memory_limit"], memoryLimit)
```

Comment:
Probably should just remove this log statement in favor of a more informative message on 695, something like:

`log.Printf("Updating memory limit from %v to %v", oldServiceData["container_memory_limit"], memoryLimit)`

Reply:
Removed, this was debug output while I was testing the original reason why I was touching the provider. All of these conditionals were actually for that failed thing, and I forgot to replace. Will undo the conditionals and run another set of integration tests